homeassistant/components/smarthab/__init__.py

```diff
@@ -53,7 +58,42 @@ def setup(hass, config) -> bool:
     # Pass hub object to child platforms
     hass.data[DOMAIN] = {DATA_HUB: hub}
 
-    load_platform(hass, "light", DOMAIN, None, config)
-    load_platform(hass, "cover", DOMAIN, None, config)
+    load_platform(hass, "light", DOMAIN, {}, config)
+    load_platform(hass, "cover", DOMAIN, {}, config)
 
     return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Set up config entry for SmartHab integration."""
+
+    # Assign configuration variables
+    username = entry.data.get(CONF_EMAIL)
+    password = entry.data.get(CONF_PASSWORD)
+
+    # Setup connection with SmartHab API
+    hub = pysmarthab.SmartHab()
+
+    try:
+        await hass.async_add_executor_job(hub.login, username, password)
+    except pysmarthab.RequestFailedException as ex:
+        _LOGGER.error("Error while trying to reach SmartHab API.")
```

Comment:
`raise ConfigEntryNotReady`

If you raise ConfigEntryNotReady Home Assistant will try again later when the api is back up

Reply:
Looks like this got missed